src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.PlaceholderLocal.cs

```diff
@@ -27,6 +27,7 @@ private sealed class PlaceholderLocal : LocalSymbol
             public PlaceholderLocal(Symbol containingSymbol, object identifier, TypeWithAnnotations type)
             {
                 Debug.Assert(identifier != null);
+                Debug.Assert(containingSymbol is null || containingSymbol.DeclaringCompilation is not null);
```

Comment:
>Debug.Assert(containingSymbol is null || containingSymbol.DeclaringCompilation is not null); [](http://example.com/codeflow?start=16&length=92)

Could we also assert something similar in `AbstractFlowPass` constructor?
```
Debug.Assert(_symbol is null || _symbol.DeclaringCompilation is not null);
```

Reply:
Unfortunately there are other scenarios (not EE related) when this assert is going to fail. For whatever reason ```AttributeSemanticModel``` uses the attribute type as member symbol and it is used as the symbol for region flow analysis, which makes little sense in my opinion.